cli/coverage_cmd.go

```diff
@@ -192,29 +209,49 @@ func coverage(repo *Repo) (*cvg.Coverage, error) {
 		}
 	}
 
-	var fileTokThresh float64 = 0.7
-	numIndexedFiles := 0
-	numDefs, numRefs, numRefsValid := 0, 0, 0
-	loc := 0 // lines of code
-	var uncoveredFiles []string
+	// Compute coverage from per-file data
+	type langStats struct {
+		numFiles        int
+		numIndexedFiles int
+		numDefs         int
+		numRefs         int
+		numRefsValid    int
+		uncoveredFiles  []string
+		loc             int
+	}
+	stats := make(map[string]*langStats)
 	for file, datum := range codeFileData {
-		loc += datum.LoC
-		numDefs += datum.NumDefs
-		numRefs += datum.NumRefs
-		numRefsValid += datum.NumRefsValid
+		if _, exist := stats[datum.Language]; !exist {
+			stats[datum.Language] = &langStats{}
+		}
+		if filepath.Base(file) == "doc.go" {
```

Comment:
I'd refactor this block and extract a function that determines if file should be taken into account